src/bin/cql-stress-cassandra-stress/settings/param/multi_param.rs

```diff
@@ -0,0 +1,240 @@
+        for param in self.subparams.iter() {
+            let mut borrowed = param.borrow_mut();
+            match borrowed.try_match(arg) {
+                ParamMatchResult::NoMatch => (),
+                ParamMatchResult::Error(e) => return ParamMatchResult::Error(e),
+                ParamMatchResult::Match => match borrowed.parse(arg) {
+                    Ok(()) => return ParamMatchResult::Match,
+                    Err(e) => return ParamMatchResult::Error(e),
+                },
+            }
+        }
+
+        ParamMatchResult::NoMatch
+    }
+
+    fn try_parse_arbitrary(&mut self, arg: &str) -> Result<()> {
+        // Ensure that argument matches pattern "key=value"
+        anyhow::ensure!(
+            ARBITRARY_PARAM.is_match(arg),
+            "Invalid {} specification: {}",
+            self.prefix,
+            arg
+        );
+
+        let split = arg.split('=').collect::<Vec<&str>>();
+        anyhow::ensure!(split.len() == 2, "Invalid arbitrary parameter: {}", arg);
+
+        let map = self.arbitrary_params.as_mut().unwrap();
+        let key = split[0].to_string();
+        let val = split[1].to_string();
+        anyhow::ensure!(
+            !map.contains_key(&key),
+            "{} suboption has been specified more than once",
+            key
+        );
+        map.insert(key, val);
+
+        Ok(())
+    }
+}
+
+impl Param for MultiParam {
+    fn parse(&mut self, arg: &str) -> Result<()> {
+        self.supplied_by_user = true;
+        let arg_val = &arg[self.prefix.len()..];
+
+        // "(some_arg)" -> "some_arg"
+        let arg_val = &arg_val[1..arg_val.len() - 1];
+
+        for (_, [subparam]) in SUBPARAM.captures_iter(arg_val).map(|m| m.extract()) {
+            // Check if the argument matches on of the predefined subparameters.
+            match self.try_parse_predefined(subparam) {
+                ParamMatchResult::Error(e) => return Err(e),
+                ParamMatchResult::Match => continue,
+                _ => (),
+            }
+
+            // If any of the subparamters didn't match, try to parse as arbitrary parameter (if applicable).
+            anyhow::ensure!(
+                self.accepts_arbitrary(),
+                "Can't accept parameter {}. {} command/option doesn't accept arbitrary parameters.",
+                subparam,
+                self.prefix
+            );
+
+            self.try_parse_arbitrary(subparam)?;
+        }
+
+        Ok(())
+    }
+
+    fn supplied_by_user(&self) -> bool {
+        self.supplied_by_user
+    }
+
+    fn required(&self) -> bool {
+        self.required
+    }
+
+    fn set_satisfied(&mut self) {
+        self.satisfied = true;
+        for param in self.subparams.iter() {
+            param.borrow_mut().set_satisfied();
+        }
+        // Clear the subparameters so the user can consume them via corresponding handles.
+        // Otherwise, .get() method called on their handles would cause panic.
+        self.subparams.clear();
```

Comment:
"Clear" in the comment isn't descriptive enough to fully explain that by removing `ParamCell`s those values can then be safely used (it isn't explained that `get()` refers to `RefCell::get()`).